### PR TITLE
sched: remove the unused function nx_wait and nx_waitid

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
This partially reverts commit de2a9d8a771ac4434d94a654e2b60c4fd0de19a1.

## Impact
Remove the unused nx_wait[id]

## Testing
Pass CI

### CODE REVIEW

File: sched/sched/sched_waitpid.c [hunk 1]
```diff
@@ -445,164 +586,24 @@ pid_t nx_waitpid(pid_t pid, int *stat_loc, int options)
...
-  /* On success, return the PID */
-
-  ret = pid;
+#ifdef CONFIG_SMP
+  leave_critical_section(flags);
+#else
+  sched_unlock();
+#endif
+  leave_cancellation_point();
+  return (int)pid;
```
Comment (pkarashchenko) [inline sched/sched/sched_waitpid.c:595]:
```suggestion
  return pid;
```

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Seems that kernel part still use it
```
 task/task_spawn.c: In function 'task_spawn':
Error: task/task_spawn.c:425:9: error: implicit declaration of function 'nx_waitpid'; did you mean 'waitpid'? [-Werror=implicit-function-declaration]
  425 |   ret = nx_waitpid(proxy, &status, 0);
      |         ^~~~~~~~~~
      |         waitpid
```

File: sched/sched/sched_waitpid.c (hunk 1 above)
Comment (xiaoxiang781216) [reply sched/sched/sched_waitpid.c]: Done.

**Comment by xiaoxiang781216** [comment]
> Seems that kernel part still use it
> 
> ```
>  task/task_spawn.c: In function 'task_spawn':
> Error: task/task_spawn.c:425:9: error: implicit declaration of function 'nx_waitpid'; did you mean 'waitpid'? [-Werror=implicit-function-declaration]
>   425 |   ret = nx_waitpid(proxy, &status, 0);
>       |         ^~~~~~~~~~
>       |         waitpid
> ```

Fixed.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**